lib/prerequisites_tree_page.rb

```diff
@@ -116,102 +80,115 @@ def prerequisite_type(prerequisite_node)
     end
   end
 
-  def prerequisite_tree(prerequisite_node)
+  def credits_prerequisite_details(prerequisite_node, group: false)
+    {
+      type: 'credits',
+      credits: node_content_from_node(prerequisite_node).split(' créditos')[0].to_i,
+    }.merge!(group ? { group: node_content_from_node(prerequisite_node).split('Grupo: ')[1].to_i } : {})
+  end
+
+  def logical_prerequisite_leaf_node_details(prerequisite_node, operator:)
+    operands =
+      extract_subjects_from_logical_prerequisite_leaf_node(prerequisite_node).each_with_object([]) do |subject, array|
+        array << {
+          type: 'subject',
+          subject_needed_code: subject[:subject_needed_code],
+          subject_needed_name: subject[:subject_needed_name],
+          needs: subject[:needs],
+        }
+      end
+
+    details = {}
+    details[:type] = 'logical'
+    details[:logical_operator] = operator
+    details[:amount_of_subjects_needed] = amount_of_subjects_needed(prerequisite_node) if operator == 'at_least'
+    details[:operands] = operands
+
+    details
```

Comment:
I originally went with:

```ruby
{
  type: 'logical',
  logical_operator: operator,
  operands: operands,
}.merge(operator == 'at_least' ? { amount_of_subjects_needed: amount_of_subjects_needed(prerequisite_node) } : {})
```

but that ended up affecting the order of the keys in `scraped_prerequisites.yml` so I took this approach instead.